### PR TITLE
[WIP] SC-86 ~ Aufgabe erstellen bug wenn kein datum gesetzt

### PR DESCRIPTION
**Pull request by adrianjost**

### CODE REVIEW

File: controllers/homework.js [hunk 1]
```diff
@@ -79,6 +79,12 @@ const getCreateHandler = (service) => {
             if ((!req.body.lessonId) || (req.body.lessonId && req.body.lessonId.length <= 2)) {
                 req.body.lessonId = null;
             }
+            if(req.body.dueDate == "__.__.____ __:__"){
```
Comment (janrenz) [inline controllers/homework.js]: can we make this more generci, for example if parsing the date fails this will be undefined

Comment (adrianjost) [reply controllers/homework.js]: You are right, I've checked again and saw, that we already have such an fallback. But for any reason the fallback is not working on the production system. If I create an task on my local test server with the master branch without any date set everything works fine. If I do the same thing on schul-cloud.org I get the alert "Das Beginndatum muss vor dem Abgabedatum liegen!".
Do you have any idea why?

**Comment by janrenz** [comment]
OK, so this is just cleaning up a new line and removing the console.log ?

**Comment by adrianjost** [comment]
jup, I haven't found the real bug. For now, I marked this pull as WIP.

Do you have any idea why the [existing fallback](https://github.com/schul-cloud/schulcloud-client/blob/46e88eb46650474ebed94ec93bcc51a572ddaa45/controllers/homework.js#L92) isn't working on the production system? On my local development setup everything works fine.

**Comment by janrenz** [comment]
Different locale settings?

**Comment by adrianjost** [comment]
What do you mean?

---------------

Currently I'm guessing that the problem is in [line 85](https://github.com/schul-cloud/schulcloud-client/blob/46e88eb46650474ebed94ec93bcc51a572ddaa45/controllers/homework.js#L85) and [line 89](https://github.com/schul-cloud/schulcloud-client/blob/46e88eb46650474ebed94ec93bcc51a572ddaa45/controllers/homework.js#L89). I think that moment isn't acting the same on my device as on the server environment when parsing an invalid date. I added a specific fallback value and hope this works. Maybe someone else can try to reproduce the original error on his device and then test if this fix works? I can only guess what's wrong because I can't reproduce the error on my machine.

Maybe @pinussilvestrus ? Here is the [ticketlink](https://ticketsystem.schul-cloud.org/browse/SC-86)

**Comment by janrenz** [comment]
different node or moment version?

**Comment by adrianjost** [comment]
what node version are we using in production?

Lars found out that the problem only occurs on linux.

**Comment by pinussilvestrus** [comment]
@adrianjost any updates regarding to this?

**Comment by adrianjost** [comment]
@pinussilvestrus I'm a Windows user and can't reproduce this bug. So it should be checked by any Linux user. For now I have no idea why this happens.

**Comment by adrianjost** [comment]
I tested it right now and the bug seems to be solved.

**Pull request closed without merging**